test_ignores and test_convert_strs fail on ppc64

Hi,

in openSUSE packaging we experience two test fails on the  ppc64 architecture (only):

```
=================================== FAILURES ===================================
_________________________________ test_ignores _________________________________

    def test_ignores():
        expected = [
            (2, ''),
            (5, ''),
            (8, '[excuse]'),
            (9, '=excuse'),
            (10, ' [excuse]'),
            (11, ' whatever'),
        ]
    
        for version in range(MIN_VER, NEXT_VER):
            tree = _ast3._parse(ignores, "<ignores>", "exec", version)
            assert [(ti.lineno, ti.tag) for ti in tree.type_ignores] == expected
            with pytest.raises(SyntaxError):
                _ast3._parse("pass  # type: ignoreé\n", "<ignores>", "exec", version)
    
    
        tree = _ast27.parse(ignores, "<ignores>", "exec")
>       assert [(ti.lineno, ti.tag) for ti in tree.type_ignores] == expected
E       AssertionError: assert [] == [(2, ''),\n (5, ''),\n (8, '[excuse]'),\n (9, '=excuse'),\n (10, ' [excuse]'),\n (11, ' whatever')]
E         Right contains 6 more items, first extra item: (2, '')
E         Full diff:
E           [
E         -  ,
E         +  (2,
E         +   ''),
E         +  (5,
E         +   ''),
E         +  (8,
E         +   '[excuse]'),
E         +  (9,
E         +   '=excuse'),
E         +  (10,
E         +   ' [excuse]'),
E         +  (11,
E         +   ' whatever'),
E           ]

ast3/tests/test_basics.py:223: AssertionError
______________________________ test_convert_strs _______________________________

    def test_convert_strs():
        ast = _ast27.parse(basic_py2, "<basic_py2>", "exec")
        tree = typed_ast.conversions.py2to3(ast)
>       assert tree.body[0].value.kind == ""
E       IndexError: list index out of range

ast3/tests/test_basics.py:305: IndexError
========================= 2 failed, 13 passed in 0.45s =========================
```

[Full build log attached](https://github.com/python/typed_ast/files/4639031/typed-ast-ppc64.log)

Maybe you have an idea what's going on? We disabled the two tests for unblock other package builds as temporary solution.

--- a/ast27/Include/asdl.h
+++ b/ast27/Include/asdl.h
@@ -28,11 +28,11 @@ typedef struct {
 } asdl_int_seq;
 
 #if PY_MINOR_VERSION > 3
-#define asdl_seq_new _Py_asdl_seq_new
-#define asdl_int_seq_new _Py_asdl_int_seq_new
+#define asdl_seq_new _Ta27_asdl_seq_new
+#define asdl_int_seq_new _Ta27_asdl_int_seq_new
 #else
-#define _Py_asdl_seq_new asdl_seq_new
-#define _Py_asdl_int_seq_new asdl_int_seq_new
+#define _Ta27_asdl_seq_new asdl_seq_new
+#define _Ta27_asdl_int_seq_new asdl_int_seq_new
 #endif
 asdl_seq *asdl_seq_new(Py_ssize_t size, PyArena *arena);
 asdl_int_seq *asdl_int_seq_new(Py_ssize_t size, PyArena *arena);

--- a/ast27/Include/asdl.h
+++ b/ast27/Include/asdl.h
@@ -27,13 +27,8 @@ typedef struct {
     int elements[1];
 } asdl_int_seq;
 
-#if PY_MINOR_VERSION > 3
 #define asdl_seq_new _Ta27_asdl_seq_new
 #define asdl_int_seq_new _Ta27_asdl_int_seq_new
-#else
-#define _Ta27_asdl_seq_new asdl_seq_new
-#define _Ta27_asdl_int_seq_new asdl_int_seq_new
-#endif
 asdl_seq *asdl_seq_new(Py_ssize_t size, PyArena *arena);
 asdl_int_seq *asdl_int_seq_new(Py_ssize_t size, PyArena *arena);
 

--- a/ast27/Python/asdl.c
+++ b/ast27/Python/asdl.c
@@ -2,7 +2,7 @@
 #include "../Include/asdl.h"
 
 asdl_seq *
-_Py_asdl_seq_new(Py_ssize_t size, PyArena *arena)
+_Ta27_asdl_seq_new(Py_ssize_t size, PyArena *arena)
 {
     asdl_seq *seq = NULL;
     size_t n;
@@ -33,7 +33,7 @@ _Py_asdl_seq_new(Py_ssize_t size, PyArena *arena)
 }
 
 asdl_int_seq *
-_Py_asdl_int_seq_new(Py_ssize_t size, PyArena *arena)
+_Ta27_asdl_int_seq_new(Py_ssize_t size, PyArena *arena)
 {
     asdl_int_seq *seq = NULL;
     size_t n;

--- a/ast27/Python/ast.c
+++ b/ast27/Python/ast.c
@@ -292,7 +292,7 @@ Ta27AST_FromNode(const node *n, PyCompilerFlags *flags, const char *filename,
             ch = CHILD(n, NCH(n) - 1);
             REQ(ch, ENDMARKER);
             num = NCH(ch);
-            type_ignores = _Py_asdl_seq_new(num, arena);
+            type_ignores = _Ta27_asdl_seq_new(num, arena);
             if (!type_ignores)
                 goto error;
 
@@ -368,7 +368,7 @@ Ta27AST_FromNode(const node *n, PyCompilerFlags *flags, const char *filename,
                         num++;
                 }
 
-                argtypes = _Py_asdl_seq_new(num, arena);
+                argtypes = _Ta27_asdl_seq_new(num, arena);
 
                 j = 0;
                 for (i = 0; i < NCH(ch); i++) {
@@ -381,7 +381,7 @@ Ta27AST_FromNode(const node *n, PyCompilerFlags *flags, const char *filename,
                 }
             }
             else
-                argtypes = _Py_asdl_seq_new(0, arena);
+                argtypes = _Ta27_asdl_seq_new(0, arena);
 
             ret = ast_for_expr(&c, CHILD(n, NCH(n) - 1));
             if (!ret)

